mambaSharedFramework/HLS Models/Playlist Structure/HLSPlaylistStructure.swift

```diff
@@ -559,16 +577,26 @@ fileprivate struct HLSPlaylistStructureConstructor {
         
         // close out our last tag
         if let startKeyIndex = startKeyIndex, let startKeyTag = startKeyTag {
-            mediaSpans.append(TagSpan(parentTag: startKeyTag, tagMediaSpan: startKeyIndex...(currentIndex - 1)))
+            let spanEnd = currentIndex - 1
+            if startKeyIndex <= spanEnd {
+                mediaSpans.append(TagSpan(parentTag: startKeyTag, tagMediaSpan: startKeyIndex...spanEnd))
+            } else {
+                assertionFailure("Invalid final media span range: \(startKeyIndex)...\(spanEnd)")
+                throw ParseError.invalidMediaSpanRange(start: startKeyIndex, end: spanEnd)
+            }
+        }
+
+        // instead of assert, warn softly if key counts mismatch (footer keys or malformed playlists)
+        if keyCount != keyTags.count {
+            print("Warning: generateMediaSpans counted \(keyCount) EXT-X-KEY tags, but found \(keyTags.count). Possibly due to footer-only key tags.")
```

Comment:
mamba doesn't really use `print`s (or other logging) so I think for consistency we'd prefer not to do this.

And why remove the `assert`? If we are seeing something strange we'd like to catch it in `debug`? `assert` symbols get stripped out for release builds.